### PR TITLE
begin splitting d_a_alink

### PR DESCRIPTION
**Pull request by TakaRikka**
long way 2 go

### CODE REVIEW

File: src/d/d_a/d_a_alink.cpp [hunk 1]
```diff
@@ -1 +1,156 @@
...
+}
+
+// 1 missing mr instruction
+#ifdef NONMATCHING
+u32 daAlink_c::checkStageName(char const* stage){
+    u32 uvar;
+    int check_stage;
+
+    check_stage = strcmp(g_dComIfG_gameInfo.current_map.stage,(char*)stage); // no idea if current stage vars are a struct or part of some class
+    uvar = __cntlzw(check_stage);
```
Comment (notyourav) [inline src/d/d_a/d_a_alink.cpp:47]: instrinsic, not a function

File: src/d/d_a/d_a_alink.cpp [hunk 2]
```diff
@@ -1 +1,156 @@
...
+    nofralloc
+    #include "d/d_a/d_a_alink/asm/func_8009DA98.s"
+}
+
+asm void daAlink_c::coHitCallback(fopAc_ac_c*, dCcD_GObjInf*){
+    nofralloc
+    #include "d/d_a/d_a_alink/asm/func_8009DB64.s"
+}
+
+asm void daAlink_c::setMatrixWorldAxisRot(float(*)[4],short,short,short,int,const cXyz*){
```
Comment (notyourav) [inline src/d/d_a/d_a_alink.cpp:68]: `const cXyz*` is inconsistent stylistically with e.g. `char const*`

File: src/d/d_a/d_a_alink.cpp [hunk 3]
```diff
@@ -1 +1,156 @@
-// ok
+#include "d/d_a/d_a_alink/d_a_alink.h"
+#include "f/f_pc/f_pc_searcher.h"
+
+u32 daAlink_c::getE3Zhint(void){
+    return 0;
+}
+
+asm char* daAlink_c::getAlinkArcName(void){
```
Comment (notyourav) [inline src/d/d_a/d_a_alink.cpp:8]: probably `const char*`

File: src/d/d_a/d_a_alink.cpp [hunk 4]
```diff
@@ -1 +1,156 @@
...
+asm void daAlink_matAnm_c::calc(J3DMaterial*) const{
+    nofralloc
+    #include "d/d_a/d_a_alink/asm/func_8009D90C.s"
+}
+
+// 1 missing mr instruction
+#ifdef NONMATCHING
+u32 daAlink_c::checkStageName(char const* stage){
+    u32 uvar;
+    int check_stage;
```
Comment (notyourav) [inline src/d/d_a/d_a_alink.cpp:44]: why mixing `int` and `u32`?

**Review by notyourav — COMMENTED** [review]
You might want to mark this PR as a draft if you plan on continue working on it.

**Review by Pheenoh — APPROVED** [review]

**Pull request merged**